### PR TITLE
REST API: Update tracking event for application password authorization web view

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #9400 

## Description

This PR adds a new property `step` to the `application_password_authorization_web_view` event to better understand user behavior when handling the authorization request. Since the authorization URL redirects to the login page, we need to track three steps:
- When the web view first opens, `step` should be `initial`.
- When the login page is displayed, `step` should be `login`. Since we cannot safely infer the admin URL of a site, the trick is to check for the `redirect_to` query.
- When the web view redirects to the authorization URL the second time, this means the login succeeds, so `step` should be `authorization`.

We will build funnels based on these steps to see where the most drop-off happen.

## Testing instructions

Prerequisite: make sure that you have access to a self-hosted store without Jetpack.
- Log out of the app if needed.
- Attempt to log in to your test store with incorrect credentials. On the error alert, select Try again with wp-admin.
- When the web view is displayed, notice in Xcode: 🔵 Tracked application_password_authorization_web_view_shown, properties: [AnyHashable("step"): "initial"].
- When the login page is shown, notice in Xcode: 🔵 Tracked application_password_authorization_web_view_shown, properties: [AnyHashable("step"): "login"].
- Log in with the correct credentials. When the authorization page is displayed, notice in Xcode console: 🔵 Tracked application_password_authorization_web_view_shown, properties: [AnyHashable("step"): "authorization"].

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by itsmeichigo** [comment]
@oguzkocer Could you include this PR in the next build of 13.0 after it gets merged, please? Thank you.

File: WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift:182]: nit: this logic seems to be based on a few assumptions briefly mentioned in the PR description, which I don't have the most thorough understanding (like why having a `redirect_to` in the URL means it's the login step, what if the store URL also contains `redirect_to` by any rare chance) 😅 maybe a comment on these assumptions can be helpful for anyone new to this code.

there's another assumption that `url.absoluteString == authorizationURL` always happens before `url.absoluteString.contains(Constants.Query.redirect)`. otherwise, the `initial` state wouldn't work.

**Review by jaclync — APPROVED** [review]
Works as expected 🚢 as mentioned in p1680854117312799/1680851919.983689-slack-C046HDZL87J, I'd recommend targeting `trunk` since release 13.0 is probably about to be released.

Just wanted to note that if the user clicks to disapprove the application password connection in the screenshot below, the `login` and `authorization` steps are skipped.

File: WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift
Comment (itsmeichigo) [reply WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift:182]: These are not assumptions, but actually how the authorization URL works, as it requires authentication and we are working with a web view that is not authenticated automatically. 

Anyway, I added comments explaining the tracking for documentation in db1738aba056907c493d9f86af58e21e10e2f807.

**Comment by itsmeichigo** [comment]
@jaclync Thanks for the review! As I responded in the Slack thread, release managers always create a final build at the end of the code freeze so I'm positive that this PR can still get in the last build if it's merged.

> Just wanted to note that if the user clicks to disapprove the application password connection in the screenshot below, the login and authorization steps are skipped.

Yes we also track the rejecting event, we plan to build the funnel from these steps to the approval.

**Pull request merged**
